src/main/scala/Head.scala

```diff
@@ -1,5 +1,27 @@
 package reinvent
 
 object Head {
-  def apply(input: List[Int]): Int = ???
+  def apply[A](input: List[A]): A = input match {
+    case Nil => throw new RuntimeException("Head of empty list")
+    case (h::_) => h
+  }
 }
+
+object Filter {
+  def apply[A](items: List[A], predicate: A => Boolean): List[A] = {
+    def applyFilter(items: List[A], accumulator: List[A]) : List[A] = {
+
+      if (items.isEmpty)  return accumulator
```

Comment:
it's rare to explicitly use the 'return' command in scala - you might find you code is clearer with pattern matching.